### PR TITLE
API changes for custom metrics

### PR DESCRIPTION
**Pull request by anushree-n**
API changes for v2.0 : Add endpoint appmetrics

### CODE REVIEW

File: info/v1/container.go [hunk 1]
```diff
@@ -58,6 +58,8 @@ type ContainerSpec struct {
 
 	// HasDiskIo when true, indicates that DiskIo stats will be available.
 	HasDiskIo bool `json:"has_diskio"`
+
+	HasCustomMetrics bool `json:"has_custommetrics,omitempty"`
```
Comment (rjnagal) [inline info/v1/container.go]: s/has_custommetrics/has_custom_metrics/

File: info/v2/container.go [hunk 2]
```diff
@@ -74,9 +74,10 @@ type ContainerSpec struct {
 	Memory    MemorySpec `json:"memory,omitempty"`
 
 	// Following resources have no associated spec, but are being isolated.
-	HasNetwork    bool `json:"has_network"`
-	HasFilesystem bool `json:"has_filesystem"`
-	HasDiskIo     bool `json:"has_diskio"`
+	HasNetwork       bool `json:"has_network"`
+	HasFilesystem    bool `json:"has_filesystem"`
+	HasDiskIo        bool `json:"has_diskio"`
+	HasCustomMetrics bool `json:"has_custommetrics"`
```
Comment (rjnagal) [inline info/v2/container.go]: We are going to add CustomMetricsSpec later though? With metric name and type, so metrics could be discovered.

**Comment by rjnagal** [comment]
ok to test

**Comment by rjnagal** [comment]
Small nits, but looks good overall.

**Comment by anushree-n** [comment]
Made the changes, @rjnagal.

File: info/v1/container.go [hunk 3]
```diff
@@ -58,6 +58,8 @@ type ContainerSpec struct {
 
 	// HasDiskIo when true, indicates that DiskIo stats will be available.
 	HasDiskIo bool `json:"has_diskio"`
+
+	HasCustomMetrics bool `json:"has_custom_metrics,omitempty"`
```
Comment (rjnagal) [inline info/v1/container.go]: nit: no omitempty for the boolean fields.

**Comment by vmarmol** [comment]
LGTM

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**